docs/static/persistent-queues.asciidoc

```diff
@@ -1,47 +1,45 @@
-persistent queues. With persistent queues enabled, Logstash persists buffered
-events to disk instead of storing them in memory. 
+persistent queues. With persistent queues enabled, Logstash stores 
+the queue on disk.
 
 Persistent queues are also useful for Logstash deployments that require high
 throughput and resiliency. Instead of deploying and managing a message
 broker, such as Redis, RabbitMQ, or Apache Kafka, to handle a mismatch in
 cadence between the shipping stage and the relatively expensive processing
-stage, you can enable persistent queues to buffer events on disk. The queue size
-is variable and configurable, which means that you have more control over
+stage, you can enable persistent queues to buffer events on disk. The persistent queue has configurable sizing, which means that you have more control over
 managing situations that can result in back pressure to the source. See <<backpressure-persistent-queues>>. 
 
 [[persistent-queues-advantages]]
 ==== Advantages of Persistent Queues
 
 Using persistent queues:
 
-* Provides protection from losing in-flight messages when the Logstash process is shut down or the machine is restarted.
-* Handles the surge of events without having to use an external queueing mechanism like Redis or Kafka.
-* Provides an at-least-once message delivery guarantee. If Logstash is restarted
-while events are in-flight, Logstash will attempt to deliver messages stored
-in the persistent queue until delivery succeeds at least once. In other words,
-messages stored in the persistent queue may be duplicated, but not lost.
+* Provides protection from losing in-flight messages when the Logstash process is abnormally terminated.
+* Absorb bursts of events without needing an external queueing mechanism like Redis or Kafka.
+* Depending on configuration, can provide an at-least-once message
+delivery guarantee. If Logstash is restarted while events are in-flight,
+Logstash will attempt to deliver messages stored in the persistent queue until
+delivery succeeds at least once. In other words, messages stored in the
+persistent queue may be duplicated, but not lost.
 
 [[persistent-queues-limitations]]
 ==== Limitations of Persistent Queues
 
-The current implementation of persistent queues has the following limitations:
+Persistent queues solves many problems, but the following are problems not
+solved:
 
-* This version does not enable full end-to-end resiliency, except for messages
-sent to the <<plugins-inputs-beats,beats>> input. For other inputs, Logstash
-only acknowledges delivery of messages in the filter and output stages, and not
-all the way back to the input or source.
+* Inputs which do not use a request-response protocol cannot be protected from data loss. For example: tcp, udp, zeromq push+pull, and many other inputs do not have a mechanism to acknowledge receipt to the sender. Plugins such as beats and http, which *do* have a acknowledgement capability, are well protected by this queue.
```

Comment:
@acchen97 I can't tell if the Kafka input actually does this. https://github.com/logstash-plugins/logstash-input-kafka/blob/master/lib/logstash/inputs/kafka.rb#L245-L260 shows no indication that I can see that it does do this acking, but it may do it somewhere deeper inside the Kafka client.